### PR TITLE
[HPRO-416] Create new biospecimen UI view for Biobank and Scripps

### PR DESCRIPTION
**Pull request by shyamnaru**

### CODE REVIEW

File: src/Pmi/Application/HpoApplication.php [hunk 1]
```diff
@@ -96,7 +96,8 @@ protected function registerSecurity()
                 [['path' => '^/site($|\/)'], ['ROLE_USER', 'ROLE_AWARDEE']],
                 [['path' => '^/help($|\/)'], ['ROLE_USER', 'ROLE_ADMIN', 'ROLE_AWARDEE', 'ROLE_DV_ADMIN']],
                 [['path' => '^/settings($|\/)'], ['ROLE_USER', 'ROLE_ADMIN', 'ROLE_AWARDEE', 'ROLE_DV_ADMIN']],
-                [['path' => '^/.*$'], 'ROLE_USER']
+                [['path' => '^/biobank($|\/)'], 'ROLE_BIOBANK', 'ROLE_SCRIPPS'],
```
Comment (stephenyeargin) [inline src/Pmi/Application/HpoApplication.php]: Not sure, but should these two roles be in their own `[]`?

Comment (shyamnaru) [reply src/Pmi/Application/HpoApplication.php]: Yes, I should be but somehow I missed it.

File: src/Pmi/Controller/BiobankController.php [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        // Allow overriding start time to test in non-prod environments
+        if (!$app->isProd() && intval($request->query->get('days')) > 0) {
+            $startString = '-' . intval($request->query->get('days')) . ' days';
+        }
+        $startTime = new \DateTime($startString, new \DateTimeZone($app->getUserTimezone()));
+        // Get MySQL date/time string in UTC
+        $startTime->setTimezone(new \DateTimezone('UTC'));
+        $today = $startTime->format('Y-m-d H:i:s');
+
+        $review = new Review;
```
Comment (jt2k) [inline src/Pmi/Controller/BiobankController.php]: Use `new Review()` for consistency.

File: src/Pmi/Controller/ReviewController.php [hunk 3]
```diff
@@ -164,7 +41,8 @@ public function todayAction(Application $app, Request $request)
         $startTime->setTimezone(new \DateTimezone('UTC'));
         $today = $startTime->format('Y-m-d H:i:s');
 
-        $participants = $this->getTodayParticipants($app['db'], $today, $site);
+        $review = new Review;
```
Comment (jt2k) [inline src/Pmi/Controller/ReviewController.php]: Use `new Review()` for consistency.

File: src/Pmi/Review/Review.php [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+            }
+            $orders[] = $row;
+        }
+
+        return $orders;
+    }
+
+    public function getOrderStatus($row, $status)
+    {
+        $order = new Order;
```
Comment (jt2k) [inline src/Pmi/Review/Review.php]: If `$order` is only used for accessing static constants, just access them directly (e.g. `Order::ORDER_CANCEL`)

File: src/Pmi/Review/Review.php [hunk 5]
```diff
@@ -0,0 +1,178 @@
+<?php
+namespace Pmi\Review;
+
+use Pmi\Order\Order;
+use Pmi\Evaluation\Evaluation;
+
+class Review
```
Comment (jt2k) [inline src/Pmi/Review/Review.php]: Since most of these methods require the database, would it make sense to pass in the db through the constructor?

**Pull request merged**